pinot-spi/src/main/java/org/apache/pinot/spi/utils/retry/RandomDelayRetryPolicy.java

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.utils.retry;
+
+import java.util.Random;
+
+
+/**
+ * Delay policy with random delay between attempts.
+ */
+public class RandomDelayRetryPolicy extends BaseRetryPolicy {
+  private final static Random RANDOM = new Random(System.currentTimeMillis());
+  private final long _minDelayMs;
+  private final int _rangeUpper;
+  private final int _rangeLower;
+
+  public RandomDelayRetryPolicy(int maxNumAttempts, long minDelayMs, long maxDelayMs) {
+    super(maxNumAttempts);
+    _minDelayMs = minDelayMs;
+    _rangeUpper = (int) ((maxDelayMs - minDelayMs) / Integer.MAX_VALUE + 1);
+    _rangeLower = (int) ((maxDelayMs - minDelayMs) % Integer.MAX_VALUE);
+  }
+
+  @Override
+  protected long getDelayMs(int currentAttempt) {
+    return RANDOM.nextInt(_rangeUpper) * Integer.MAX_VALUE + RANDOM.nextInt(_rangeLower) + _minDelayMs;
```

Comment:
isn't it simpler to just do:
`return _minDelayMs + (long)((_maxDelayMs - _minDelayMs) * RANDOM.nextDouble())`?

Reply:
done